molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/controller/DataExplorerController.java

```diff
@@ -415,11 +408,41 @@ public String viewEntityDetails(@RequestParam(value = "entityTypeId") String ent
 
 		model.addAttribute("entity", dataService.getRepository(entityTypeId).findOneById(id));
 		model.addAttribute("entityType", entityType);
-		model.addAttribute("viewName", getViewName(entityTypeId));
+		model.addAttribute("viewName", getEntityReportViewName(entityTypeId));
+
+		// Used to create a URL to a standalone report
+		model.addAttribute("showStandaloneReportUrl", dataExplorerSettings.getModStandaloneReports());
+		model.addAttribute("entityTypeId", entityTypeId);
+		model.addAttribute("entityId", entityId);
+
 		return "view-entityreport";
 	}
 
-	private String getViewName(String entityTypeId)
+	/**
+	 * Builds a model containing one entity and returns standalone report ftl view
+	 *
+	 * @param entityTypeId
+	 * @param entityId
+	 * @param model
+	 * @return standalone report view
+	 * @throws Exception if an entity name or id is not found
+	 */
+	@RequestMapping(value = "/details/{entityTypeId}/{entityId}", method = GET)
+	public String viewEntityDetailsById(@PathVariable(value = "entityTypeId") String entityTypeId,
+			@PathVariable(value = "entityId") String entityId, Model model) throws Exception
+	{
+		EntityType entityType = dataService.getEntityType(entityTypeId);
+		Object id = getTypedValue(entityId, entityType.getIdAttribute());
+
+		model.addAttribute("entity", dataService.getRepository(entityTypeId).findOneById(id));
```

Comment:
Does not happen because the ID originates from a on row click in the UI.
The only way this fails, if somehow magically that row is deleted in the same millisecond it is clicked